wallet/wallet.go

```diff
@@ -301,6 +302,34 @@ func (w *SingleAddressWallet) Redistribute(cs consensus.State, outputs int, amou
 	w.mu.Lock()
 	defer w.mu.Unlock()
 
+	// build map of inputs currently in the tx pool
+	inPool := make(map[types.Hash256]bool)
+	for _, ptxn := range pool {
+		for _, in := range ptxn.SiacoinInputs {
+			inPool[types.Hash256(in.ParentID)] = true
+		}
+	}
+
+	// fetch unspent transaction outputs
+	utxos, err := w.store.UnspentSiacoinElements(false)
+	if err != nil {
+		return types.Transaction{}, nil, err
+	}
+
+	// check whether a redistribution is necessary, adjust number of desired
+	// outputs accordingly
+	for _, sce := range utxos {
+		inUse := w.isOutputUsed(sce.ID) || inPool[sce.ID]
+		matured := cs.Index.Height >= sce.MaturityHeight
+		sameValue := sce.Value.Equals(amount)
```

Comment:
Shouldn't we check for `>=` outputs?

Reply:
That's what I thought so too originally but I don't think we should do that. It gets a little weird because imagine you can't split this output to form another one that you are still missing. E.g. if you encounter a 25SC output and are splitting into 10SC outputs you'd have to 'reserve' the output to make sure it doesn't get used later when you're splitting it up into 10-10-5 to form outputs you might be missing.

I also figured that conceptually it's nicer that `Redistribute` just  means "ensure my wallet has x outputs of value y" instead of "at least value y". Don't mind changing this though but in practice checking for `Equals` works.